ckanext/datastore/backend/postgres.py

```diff
@@ -1901,6 +1901,7 @@ def datastore_search(
         elif fields and fts_q:
             field_ids = datastore_helpers.get_list(fields)
             all_field_ids = list(fields_types.keys())
+            all_field_ids.remove("rank")
```

Comment:
Users might rely on `rank` being returned, after all it determines the order of the returned records so there should be a way to retrieve it. 
Let's keep `rank` in the returned fields if it is defined in the incoming `fields` parameter, eg something like


```suggestion
            if "rank" not in fields:
                all_field_ids.remove("rank")
```

And please add a `changes/6961.misc` file explaining the change in behaviour (eg The "rank" field is no longer returned in `datastore_search` results unless explicitly defined in the `fields` parameter)

Reply:
@amercader , thanks for your response, will do the changes as you suggested.